cwltool/draft2tool.py

```diff
@@ -1,43 +1,56 @@
 import avro.schema
 import json
 import copy
-from flatten import flatten
+from .flatten import flatten
 import functools
 import os
-from pathmapper import PathMapper, DockerPathMapper
-from job import CommandLineJob
+from .pathmapper import PathMapper, DockerPathMapper
+from .job import CommandLineJob
 import yaml
-import glob
 import logging
 import hashlib
-import random
-from process import Process, shortname, uniquename, adjustFileObjs
-from errors import WorkflowException
+from .process import Process, shortname, uniquename, adjustFileObjs
+from .errors import WorkflowException
 import schema_salad.validate as validate
-from aslist import aslist
-import expression
-import re
-import urlparse
+from .aslist import aslist
 import tempfile
-from builder import CONTENT_LIMIT, substitute
+from .builder import CONTENT_LIMIT, substitute, Builder
 import shellescape
 import errno
+from typing import Callable, Any, Union, Generator
 
 _logger = logging.getLogger("cwltool")
 
+
 class ExpressionTool(Process):
+
     def __init__(self, toolpath_object, **kwargs):
+        # type: (Dict[str,List[None]], **Any) -> None
         super(ExpressionTool, self).__init__(toolpath_object, **kwargs)
 
     class ExpressionJob(object):
-        def run(self, **kwargs):
+
+        def __init__(self):  # type: () -> None
+            self.builder = None  # type: Builder
+            self.requirements = None  # type: Dict[str,str]
+            self.hints = None  # type: Dict[str,str]
+            self.collect_outputs = None  # type: Callable[[Any], Any]
+            self.output_callback = None  # type: Callable[[Any, Any], Any]
+            self.outdir = None  # type: str
+            self.tmpdir = None  # type: str
+            self.script = None  # type: Dict[str,str]
```

Comment:
Some of these type definitions might be able to go away (and break the dependency cycle). Still useful to have all the private attributes defined in one place for developer comprehension.